pkg/mdformatter/mdformatter_bench_test.go

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Bartłomiej Płotka @bwplotka
+// Licensed under the Apache License 2.0.
+
+package mdformatter
+
+import (
+	"bytes"
+	"context"
+	"os"
+	"testing"
+
+	"github.com/efficientgo/tools/core/pkg/testutil"
+)
+
+func benchMdformatter(filename string, b *testing.B) {
+	file, err := os.OpenFile(filename, os.O_RDONLY, 0)
+	testutil.Ok(b, err)
+	defer file.Close()
+
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		f := New(context.Background())
+
+		b.Run(filename, func(b *testing.B) {
```

Comment:
It's important to put `b.N` loop INSIDE the nested b.Run as this is the test case that will be benchmarked. Since you don't have we have misleading results I think.

Reply:
Yes! Updated results:

```
go test -run=^$ -bench=. -benchtime 20s -memprofile mem.out -cpuprofile cpu.out -benchmem github.com/bwplotka/mdox/pkg/mdformatter/mdgen          
goos: darwin
goarch: amd64
pkg: github.com/bwplotka/mdox/pkg/mdformatter/mdgen
cpu: Intel(R) Core(TM) i5-5250U CPU @ 1.60GHz
Benchmark_Mdgen_Sleep2/benchdata/sleep2.md-4              361028             65180 ns/op           42746 B/op        375 allocs/op
Benchmark_Mdgen_Sleep5/benchdata/sleep5.md-4              403152             62930 ns/op           42748 B/op        375 allocs/op
Benchmark_Mdgen_GoHelp/benchdata/gohelp.md-4              432640             57559 ns/op           42744 B/op        375 allocs/op
PASS
ok      github.com/bwplotka/mdox/pkg/mdformatter/mdgen  100.219s
```
```
go test -run=^$ -bench=. -benchtime 20s -memprofile mem.out -cpuprofile cpu.out -benchmem github.com/bwplotka/mdox/pkg/mdformatter
goos: darwin
goarch: amd64
pkg: github.com/bwplotka/mdox/pkg/mdformatter
cpu: Intel(R) Core(TM) i5-5250U CPU @ 1.60GHz
Benchmark_Mdformatter/testdata/not_formatted.md-4                 396523             61845 ns/op           42747 B/op        375 allocs/op
PASS
ok      github.com/bwplotka/mdox/pkg/mdformatter        44.476s
```